cmd/bosun/expr/funcs.go

```diff
@@ -691,11 +692,16 @@ func timeTSDBRequest(e *State, T miniprofiler.Timer, req *opentsdb.Request) (s o
 	return
 }
 
-func timeLSRequest(e *State, T miniprofiler.Timer, service *elastic.SearchService, source *elastic.SearchSource) (resp *elastic.SearchResult, err error) {
-	e.logstashQueries = append(e.logstashQueries, *service.SearchSource(source))
-	b, _ := json.MarshalIndent(source.Source(), "", "  ")
+func timeLSRequest(e *State, T miniprofiler.Timer, req *logstash.Request) (resp *elastic.SearchResult, err error) {
+	e.logstashQueries = append(e.logstashQueries, *req.Source)
+	b, _ := json.MarshalIndent(req.Source.Source(), "", "  ")
 	T.StepCustomTiming("logstash", "query", string(b), func() {
-		resp, err = service.SearchSource(source).Do()
+		getFn := func() (interface{}, error) {
+			return e.logstashContext.Query(req)
+		}
+		var val interface{}
+		val, err = e.cache.Get(string(b), getFn)
```

Comment:
Elastic Response -> MakeIntoBosunResult -> BosunResult

I'm caching the elastic response, not the BosunResult. By processing I mean
transforming an elastic response to a Bosun Result.

On Wed, Mar 25, 2015 at 2:11 PM, Dieter Plaetinck <notifications@github.com>
wrote:

> In cmd/bosun/expr/funcs.go
> <https://github.com/bosun-monitor/bosun/pull/840#discussion_r27149243>:
>
> >  	T.StepCustomTiming("logstash", "query", string(b), func() {
> > -		resp, err = service.SearchSource(source).Do()
> > +		getFn := func() (interface{}, error) {
> > +			return e.logstashContext.Query(req)
> > +		}
> > +		var val interface{}
> > +		val, err = e.cache.Get(string(b), getFn)
>
> what is the response you're caching, if not the results from the query?
> and what does "processing" mean?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bosun-monitor/bosun/pull/840/files#r27149243>.
>